### DISCUSSION TITLE
Tools and workflow?

### DISCUSSION THREAD

**Discussion by mdtrooper** [OpenRCT2/OpenGraphics#27] [Q&A] [answered]
I want to help but I don't know how start.

- Is there a tool or parameter in cli mode of OpenTTD to extract the sprites?
- Is there a tool to pack the "custom sprites" into .dat file?

Thanks.

**Comment by duncanspumpkin** [comment]
Assume you mean OpenRCT2. We have a tool that can extract all sprites from dat files or g1.dat. 

https://github.com/OpenRCT2/OpenRCT2/wiki/Sprite-compiler

You shouldn't be creating .dat files but instead .parkobj files which are just zip files with the extension changed. Inside the zip you put all your images and your object.json file that has all the meta data. I suggest joining the discord for further info.

**Comment by mdtrooper** [comment]
Well is there a list of DAT?

Because I found this https://github.com/OpenRCT2/OpenRCT2/issues/13576 .

And I found this part of code https://github.com/OpenRCT2/OpenRCT2/blob/77141f57b0a91aa4321bc6908027168fe883a784/src/openrct2/rct1/Tables.cpp .

It runs as:
```
$ openrct2 sprite exportalldat PTCT1 /tmp/test
```

And I tried with popcorn building:
```
$ openrct2 sprite exportalldat POPCS /tmp/pis/
{ "path": "/tmp/pis/0.png", "x": 0, "y": 0 },
{ "path": "/tmp/pis/1.png", "x": 0, "y": 0 },
{ "path": "/tmp/pis/2.png", "x": 0, "y": 0 },
{ "path": "/tmp/pis/3.png", "x": -31, "y": -39 },
{ "path": "/tmp/pis/4.png", "x": -31, "y": -44 },
{ "path": "/tmp/pis/5.png", "x": -32, "y": -45 },
{ "path": "/tmp/pis/6.png", "x": -32, "y": -45 },
```
But it is some hard to look for into C file.

**Comment by duncanspumpkin** [comment]
I am unsure what you are trying to do. If it's create a new park object you can view the json files of the vanilla game at https://github.com/OpenRCT2/objects the images of vanilla objects can be extracted as you have found using the sprite exporter. Then to create your new park object you just combine the json file and the images into a zip file and rename it.

**Comment by mdtrooper** [comment]
Ohh, it is great.

```
{
    "id": "rct2.ride.popcs",
    "authors": ["Chris Sawyer", "Simon Foster"],
    "version": "1.0",
    "originalId": "0A188B80|POPCS   |36E12446",
    "sourceGame": ["rct2", "rct1"],
    "objectType": "ride",
    "properties": {
        "type": "food_stall",
        "category": "stall",
        "sells": "popcorn",
        "disablePainting": true,
        "carsPerFlatRide": 1,
        "carColours": [
            [
                ["black", "black", "black"]
            ]
        ]
    },
    "images": ["$RCT2:OBJDATA/POPCS.DAT[0..6]"],
    "strings": {
        "name": {
            "en-GB": "Popcorn Stall",
            "fr-FR": "Popcorn",
            "de-DE": "Popcornstand",
            "es-ES": "Puesto de palomitas",
            "it-IT": "Chiosco dei pop-corn",
            "nl-NL": "Popcornkraam",
            "sv-SE": "Popcornstånd",
            "ko-KR": "팝콘 가게",
            "zh-CN": "爆米花店",
            "zh-TW": "爆米花店",
            "pt-BR": "Barraca de Pipoca",
            "cs-CZ": "Stánek s popcornem",
            "ja-JP": "ポップコーンショップ",
            "pl-PL": "Kiosk z popcornem",
            "ru-RU": "Попкорн",
            "eo-OO": "Krevmaizbudo"
        },
        "description": {
            "en-GB": "A stall selling giant buckets of popcorn",
            "fr-FR": "Boutique vendant de grands pots de popcorn",
            "de-DE": "Ein Stand, an dem riesige Popcorneimer verkauft werden",
            "es-ES": "Un puesto que vende gigantes cubos de palomitas",
            "it-IT": "Un chiosco che vende cesti giganti di pop-corn",
            "nl-NL": "Een kraam waar grote emmers popcorn worden verkocht.",
            "sv-SE": "Ett stånd som säljer jättebägare med popcorn",
            "ko-KR": "팝콘 한 바구니를 가득 채워 파는 가게",
            "zh-CN": "卖大桶爆米花的店",
            "zh-TW": "賣大桶爆米花的店",
            "pt-BR": "Uma barraca vendendo baldes gigantes de pipoca",
            "cs-CZ": "Stánek prodávající obrovské kelímky popcornu",
            "ja-JP": "ポップコーンを売る売店です",
            "pl-PL": "Stoisko sprzedające wielkie pudełka popcornu",
            "ru-RU": "Палатка с Попкорном",
            "eo-OO": "Budo vendanta grandajn ujojn da krevmaizo"
        }
    }
}
```

Could I do a script to generate a table or list or something in markdown with the progress? Because it will be a help to new people who want to help.

**Comment by duncanspumpkin** [comment]
At the moment there isn't really anything implemented. There are a couple of attempts in this repo but we aren't using them atm in the main openrct2 game. @oli414 is currently trying to decide how best to produce assets. I personally feel folks should just bash out a couple of object designs and from there decide.

**Comment by oli414** [comment, marked as answer]
The best way to help is to just create objects using Blender without worrying too much about the rest of the process yet :-)

Static scenery objects and shops are the best type of objects to get started on, check out the Standards and Requirements page to make sure that your contributions are compatible with the contributions made by others:
https://github.com/OpenRCT2/OpenGraphics/wiki/Standards-and-Requirements

As Duncan mentioned, in terms of tools nothing has been implemented yet at the time of writing this.

**Reply by mdtrooper** [reply]
> The best way to help is to just create objects using Blender without worrying too much about the rest of the process yet :-)

Yes, it is the point.
